detections/application/cisco_secure_application_alerts.yml

```diff
@@ -0,0 +1,88 @@
+id: 9982bff4-fc5d-49a3-ab9e-2dbbab2a711b
+version: 1
+date: '2025-02-04'
+author: Ryan Long, Bhavin Patel, Splunk
+status: production
+type: Anomaly
+description: |
+  The following analytic is to leverage alerts from Cisco SecureApp, which identifies and monitors exploit attempts targeting business applications. The primary attack observed involves exploiting vulnerabilities in web applications, including injection attacks (SQL, API abuse), deserialization vulnerabilities, remote code execution attempts, LOG4J and zero day attacks. These attacks are typically aimed at gaining unauthorized access, exfiltrating sensitive data, or disrupting application functionality.
+
+  Cisco SecureApp provides real-time detection of these threats by analyzing application-layer events and correlating attack behavior with known vulnerability signatures. This detection methodology helps the Security Operations Center (SOC) by:
+
+  * Identifying active exploitation attempts in real-time, allowing for quicker incident response.
+  * Categorizing attack severity to prioritize remediation efforts based on risk level.
+  * Providing visibility into attacker tactics, including source IP, attack techniques, and affected applications.
+  * Generating risk-based scoring and contextual alerts to enhance decision-making within SOC workflows.
+  * Helping analysts determine whether an attack was merely an attempt or if it successfully exploited a vulnerability.
+
+  By leveraging this information, SOC teams can proactively mitigate security threats, patch vulnerable applications, and enforce security controls to prevent further exploitation.
+data_source:
+- Cisco Secure Application AppDynamics Alerts
+search: |-
+  `appdynamics_security` blocked=false 
+  | rename attackEvents{}.attackOutcome AS attackOutcome, "attackEvents{}.vulnerabilityInfo.*" AS * 
+  | fields - tag::eventtype, eventtype, host, id, index, linecount, punct, source, sourcetype, splunk_server, tag, SourceType, app clientAddressType, application, tier, "attackEvents{}.*" 
+  | eval socketOut=mvjoin(socketOut," AND ") 
+  | eval risk_score=kennaScore 
+  | fillnull risk_score value="0" 
+  | eval risk_object=app_name 
+  | stats values(*) as * by attackId 
+  | eval severity=case(
+      risk_score>=100 OR signature="LOG4J", "critical",
+      risk_score>50 AND risk_score<75, "high",
+      risk_score=0 AND attackOutcome="EXPLOITED", "high",
+      risk_score<=50 AND attackOutcome!="OBSERVED", "medium",
+      risk_score=0 AND attackOutcome="ATTEMPTED", "medium",
+      risk_score=0, "low",
+      risk_score=0 AND attackOutcome="OBSERVED", "low"
+      ) 
+  | eval risk_message=case(
+      (signature="API" OR signature="LOG4J" OR signature="SSRF"), "An ".attackOutcome." ".signature." vulnerability is attempted to be abused from ".src_category." IP address ".src_ip." and was seen connecting to server ".dest_nt_host." hosting application ".app_name." and possibly exfiltrating data to ".socketOut."",
```

Comment:
The wording/grammar here seems off when you substitute values present in the above case. 
> "An OBSERVED LOG4J vulnerability is attempted to be abused" 

for example. I think at minimum s/is/was, but the wording still doesn't seem right.

Reply:
Updated the syntax for this now the output is :

Example: 

```
The application AD-Ecommerce deserializes untrusted data without sufficiently verifying that the resulting data will be valid. Data which is untrusted cannot be trusted to be well-formed. Malformed data or unexpected data could be used to abuse application logic, deny service, or execute arbitrary code, when deserialized.

An attempt to exploit a LOG4J vulnerability was made from a EXTERNAL IP address 198.26.180.33. The server /usr/src/argento/prod/demo-run/tomcat-demo-app/webapps/argentoDemoApp/ hosting application AD-Ecommerce was accessed, and data may have been exfiltrated to 212.160.116.247:443 AND 220.213.253.20:443.

```